torchdata/datapipes/iter/util/header.py

```diff
@@ -60,3 +61,37 @@ def __len__(self) -> int:
                 "The actual value may be smaller if the actual length of source_datapipe is smaller than the limit."
             )
             return self.limit
+
+
+@functional_datapipe("set_length")
```

Comment:
What would be the difference between adding this DataPipe and implement `set_length` API to the base class of `IterDataPipe`?